oci_image.py

```diff
@@ -27,12 +28,24 @@ def fetch(self):
         else:
             # Translate the data from the format used by the charm store to the
             # format used by the Juju K8s pod spec, since that is how this is
-            # typically used.
-            return {
-                'imagePath': resource_data['registrypath'],
-                'username': resource_data['username'],
-                'password': resource_data['password'],
-            }
+            # typically used:
```

Comment:
Is this change ok for you @johnsca? In case no `username` nor `password` are given, instead of 

```
{
    'imagePath': image,
    'password': '',
    'username': ''
}
```
Will be
```
{
    'imagePath': image,
}
```